tools/integration-test/src/tests/client_upgrade.rs

```diff
@@ -0,0 +1,577 @@
+//! This test tests four different cases:
+//!
+//! - The `ClientUpgradeTest` tests the case where the client upgrade works
+//!   correctly after the chain was upgraded.
+//!
+//! - The `InvalidClientUpgradeTest` tests the case where the
+//!   client upgrade fails as the chain has not been upgraded.
+//!
+//! - The `HeightTooHighClientUpgradeTest`tests the case where the client
+//!   fails to upgrade because a height too big is given as input.
+//!
+//! - The `test_height_too_low_client_upgrade`tests the case where the client
+//!   fails to upgrade because a height too small is given as input.
+
+use http::Uri;
+use prost::Message;
+use std::str::FromStr;
+
+use ibc_proto::cosmos::gov::v1beta1::{query_client::QueryClient, QueryProposalRequest};
+use ibc_proto::ibc::core::client::v1::UpgradeProposal;
+use ibc_relayer::error::Error as RelayerError;
+use ibc_relayer::upgrade_chain::{build_and_send_ibc_upgrade_proposal, UpgradePlanOptions};
+use ibc_relayer_types::core::ics02_client::height::Height;
+use ibc_test_framework::{chain::cli::upgrade::vote_proposal, prelude::*};
+
+const MAX_DEPOSIT_PERIOD: &str = "10s";
+const VOTING_PERIOD: &str = "10s";
+const DELTA_HEIGHT: u64 = 15;
+const WAIT_CHAIN_UPGRADE: Duration = Duration::from_secs(4);
+
+#[test]
+fn test_client_upgrade() -> Result<(), Error> {
+    run_binary_chain_test(&ClientUpgradeTest)
+}
+
+#[test]
+fn test_invalid_client_upgrade() -> Result<(), Error> {
+    run_binary_chain_test(&InvalidClientUpgradeTest)
+}
+
+#[test]
+fn test_height_too_high_client_upgrade() -> Result<(), Error> {
+    run_binary_chain_test(&HeightTooHighClientUpgradeTest)
+}
+
+#[test]
+fn test_height_too_low_client_upgrade() -> Result<(), Error> {
+    run_binary_chain_test(&HeightTooLowClientUpgradeTest)
+}
+
+struct ClientUpgradeTest;
+
+impl TestOverrides for ClientUpgradeTest {
+    /// Update the genesis file in order to reduce the time required to upgrade the chain.
+    fn modify_genesis_file(&self, genesis: &mut serde_json::Value) -> Result<(), Error> {
+        use serde_json::Value;
+
+        let max_deposit_periods = genesis
```

Comment:
We can define a helper function in `tools/test-framework/src/chain/config.rs` or similar location to modify the config parameters.

Reply:
This has been addressed in this commit https://github.com/informalsystems/hermes/pull/2795/commits/88e15de2a355dce396d45ed1688db9b2d12f6a6e

Reply:
Did you missed updating the use of the update method here?

Reply:
It's the single implementation for `ClientUpgradeTestOverrides`, I could move it at the bottom if it avoids confusions. What do you think ?

Reply:
It is fine to move it to the config helper even if it is only used once. Functions are used for abstractions and improving readability, not to remove code duplication.

Reply:
I'm sorry, I misunderstood your comment. I had forgotten to remove the duplicate code updating the max_deposit_period. It has been removed in 00d1529dfc2a47d6b06a702ba0795f93691f8eb9